---
title: Investigating sex differences in genetic diversity in great-tailed grackles in Tempe, Arizona to infer potential sex biases in dispersal
author: 'August Sevchik (Arizona State University), [Dr. Corina Logan](http://CorinaLogan.com) (Max Planck Institute for Evolutionary Anthropology, MPI EVA, corina_logan@eva.mpg.de), Melissa Folsom (MPI EVA), Luisa Bergeron (University of California Santa Barbara), [Dr. Aaron Blackwell](https://blackwell-lab.com) (Washington State University), Carol Rowney (MPI EVA), [Dr. Dieter Lukas](http://dieterlukas.strikingly.com) (MPI EVA, dieter.lukas@eva.mpg.de)'
date: '`r Sys.Date()`'
output:
  html_document: 
    toc: true
    toc_depth: 4
    toc_float: 
      collapsed: false
    code_folding: hide 
  md_document: 
    toc: true
  pdf_document:
    keep_tex: yes
    latex_engine: xelatex
  github_document: 
    toc: true
bibliography: /Users/corina/GitHub/grackles/Files/MyLibrary.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#Make code wrap text so it doesn't go off the page when Knitting to PDF
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
```

***Click [here](https://github.com/corinalogan/grackles/blob/master/README.md) to navigate to the version-tracked reproducible manuscript (.Rmd file)***

###ABSTRACT

In most bird species, females are the dispersing sex, while males remain close to the place they were hatched for their whole lives (@greenwood1980mating). Explanations for such female bias in dispersal have focused on the potential benefits that males derive from knowing the local environment to establish territories, while females search for suitable mates (@greenwood1980mating), however the exact factors shaping dispersal decisions appear more complex (@mabry2013social, @vegvari2018sex). Here, we investigate whether females are the dispersing sex in great-tailed grackles, which have a mating system where the males hold territories and the females choose which territory to place their nest on (@johnson2000male). We will use genetic approaches to identify sex biases in the propensity to disperse and to potentially infer the distance that individuals move. Results will have bearing on our long-term study on the relationship between behavioral flexibility and rapid geographic range expansions that is being conducted on this species by allowing us to understand which individuals are likely to experience similar conditions across their lives and which are likely to face new conditions when they become breeders.

###A. STATE OF THE DATA

This preregistration was written (June 2019) and submitted to Peer Community In Ecology (19 July 2019) after blood was collected and before processing the DNA to obtain the genetic data.

###B. HYPOTHESIS

There are sex differences in the distribution of great-tailed grackle (*Quiscalus mexicanus*) genetic diversity as a result of differential dispersal rates of females versus those of males. 

**Prediction 1** Average genetic relatedness will be lower among females compared to males amoung our DNA samples. Males are expected to remain close to the area where they hatched, therefore all of the males on the Arizona State University (ASU) campus are expected to come from the same population. In contrast, females are expected to move before their first breeding attempt and therefore the females on campus are likely to come from multiple populations in the broader area because in birds females tend to be the dispersing sex (@greenwood1980mating).

**Prediction 1 alternative 1:** Genetic diversity is higher and in turn average relatedness is lower in males than in females, potentially because males move more than females in this species, or, even though fewer males may move between populations, if they do, they move much larger distances and are therefore more likely to bring in divergent genetic variants.

**Prediction 1 alternative 2:** Males and females move equal distances and therefore their genetic diversity and average relatedness are similar. 

**Prediction 1 alternative 3:** The distances that individuals move are smaller or larger than the area of the ASU campus and therefore our samples do not represent the broader population. If the distance moved is within the scale of the campus area, then only very few individuals will have come from outside the campus area and have brought new genetic variants. If the distance moved is broader than the scale of the campus area, then the campus might include multiple populations and therefore males might also be from different populations containing different genetic variants.

**Prediction 2:** There will be a negative correlation between the distance at which two males were trapped and their genetic relatedness because close kin, such as brothers, will remain on the same or on neighboring territories. Whereas, for females, we do not expect any association between spatial and genetic distance because in birds females tend to be the dispersing sex so that relatives will be found at varying distances from each other (Greenwood 1980). 

**Prediction 2 alternative 1:** There will be no relationship between the distance at which two males were trapped and their genetic relatedness. There are a couple of potential explanations for why this would occur and we are not able to distinguish between them: 1) the distances between trap sites are not large enough to detect a decline in relatedness, 2) males move away from their hatch sites to breed, or 3) the population might have such low genetic diversity that even unrelated males tend to share many genetic variants.

###C. METHODS

####**Planned Sample**
	
DNA from 53 great-tailed grackles was obtained from wild adults and juveniles (n=38 adult females, n=15 adult males, juvenile samples were excluded because they had not yet dispersed) caught in Tempe, Arizona, USA. These individuals were either immediately released, or temporarily brought into aviaries for behavioral testing and then released back to the wild.
	
####**Sample size rationale**
	
This was the largest sample size possible by July 2019 when the DNA were analyzed using ddRADseq. 
	
####**Data collection stopping rule**
	
Analyze all blood that was collected by June 2019 when the trapping season stopped.

####**Open data**

When the study is complete, the data will be published in the Knowledge Network for Biocomplexity's data repository.

####**Randomization and counterbalancing** 

No randomization or counterbalancing is involved in this study.

####**Blinding of conditions during analysis**

Experimenters are blind to the sex of the bird when processing samples using ddRADseq (only the alphanumeric bird ID was visible on the tube and no team member has memorized which ID goes with which bird because we give the birds names).

####**Blood collection**

150uL of blood was collected from trapped great-tailed grackles and stored in 650uL lysis buffer (@white1992mitochondrial, pp. 50-51) for a minimum of 30 minutes, and a maximum of 60 minutes at room temperature or a maximum of 3 hours on ice. Samples were then centrifuged at 15 x g for 10 minutes to separate the serum from the red blood cells. After the serum layer was removed from the centrifuged samples, 600uL lysis buffer was added to the remaining packed red blood cells. These tubes containing packed red blood cells and lysis buffer were then stored at room temperature until the DNA was extracted.

####**DNA extraction and quantification**

DNA was extracted from the above samples using the DNeasy Blood and Tissue kit (Qiagen) with slight modifications from the manufacturer's protocol (see details in @thrasher2018double Supporting Information, page 7). DNA quantification was then performed on a Qubit 4.0 Fluorometer (Fisher Scientific) following the manufacturer’s protocol.

####**ddRADseq**

The DNA will be processed using ddRADseq by Sevchik and Bronwyn Butcher (Cornell University) following methods in @thrasher2018double. In brief, DNA extracts are digested with a combination of two restriction enzymes, and 20 different adapters attached to the end of the DNA pieces. Fragments of a prespecified length (between 450-600 base pairs) will be selected after purification. Index groups and sequencing adapters will be added by performing PCR. Sequencing will be performed on an Illumina NextSeq500 (using a mid-output kit) to generate 150 bp single end reads. These data are post-processed to generate SNP data for relatedness analyses as in @thrasher2018double. After filtering reads for quality and demultiplexing to assign sequences back to specific individuals, RAD loci will be assembled de novo as now reference genome exists for great-tailed grackles. We will follow the cut-offs described in @thrasher2018double for SNP filtering, but might adjust settings (e.g. minor allele frequencies) depending on the composition of our sample (these decisions will have no influence on testing our predictions and will be performed prior to any further analyses). 

####**Relatedness analyses**

Genetic relatedness between all pairs of individuals is calculated using the package “related” (Pew, Muir, Wang, & Frasier, 2015 https://doi.org/10.1111/1755-0998.12323) in R, following methods in @thrasher2018double. We will use the function 'compareestimators' to assess which relatedness estimator appears to perform the best given the characteristics of our data. 

####**Dependent variable**

Average relatedness between all pairs of individuals

####**Independent variables**

P1: Sex (female, male)

P2: Distance between trap sites (meters)

###D. ANALYSIS PLAN

We do not plan to **exclude** any data and no data are **missing**. Analyses will be conducted in R (current version `r getRversion()`; @rcoreteam). 

####*Ability to detect actual effects*

The number of males in our sample might be too low to detect patterns of isolation-by-distance (change in relatedness within the range of our sampling locations) or to infer the distance of dispersal. Having samples only from a single population might limit our ability to determine frequencies of dispersal, whether just some or the majority of the individuals of one sex or the other disperse. The sample should, however, be sufficient to obtain a qualitative assessment of whether dispersal is more prevalent in either females or males. 

####*P1: average relatedness and sex*

**Analysis:** 

We will compare the mean and variance in average relatedness among all females to that among all males. Since average relatedness tends to decrease as the number of individuals in the sample increases (regression to the mean), we will perform a permutation analysis to investigate whether the average relatedness among the males in our sample is higher than what would be expected for a random sample of the same number of females or of individuals of both sexes. We will repeat 10,000 random draws of 15 individuals either from among the females or from among all individuals, and generate distributions of average relatedness among these samples. We wil assess how the observed average relatedness among the 15 males compares to these random samples.  

####*P2: average relatedness and trap site distance*

**Analysis:** 

To test whether males and females show different isolation-by-distance, we will follow analyses as in @aguillon2017deconstructing. For the analysis, we will initially create 11 distance bins separated by 50m between 0m-500m (the maximum distance between trapping sites). The 50m bin size was chosen because there are roosting trees that are ~50m apart suggesting that dispersal might be occuring at this scale. The individuals in our sample have been caught at one of 15 trap sites, and the resulting 105 pairwise distances among individuals will be assigned to one of the 11 bins. We will reduce the number of bins in case the pairwise distances among individuals cluster at specific distances. For males and females separately, we will link the matrices of average relatedness and of distance between all pairs of individuals by first plotting genetic relatedness against geographic distance and next by assessing the strength of the association using Mantel correlograms. We will use the function 'mantel.correlog' in the vegan package (Oksanen J, Blanchet FG, Kindt R, Legendre P, Minchin PR, O'Hara RB, et al. vegan: Community Ecology Package. 2015) in R, performing 10,000 permutations to assess the strength of the association. A stronger negative correlation between genetic relatedness and spatial distance for males than for females would indicate that males disperse shorter distances than females, and in particular we expect that males captured at the same trapping site will be much closer related to each other than females captured at the same trapping site. 

###E. ETHICS

This research is carried out in accordance with permits from the:

1) US Fish and Wildlife Service (scientific collecting permit number MB76700A-0,1,2)
2) US Geological Survey Bird Banding Laboratory (federal bird banding permit number 23872)
3) Arizona Game and Fish Department (scientific collecting license number SP594338 [2017], SP606267 [2018], and SP639866 [2019])
4) Institutional Animal Care and Use Committee at Arizona State University (protocol number 17-1594R)
5) University of Cambridge ethical review process (non-regulated use of animals in scientific procedures: zoo4/17 [2017])

###F. AUTHOR CONTRIBUTIONS

**Sevchik:** Hypothesis development, sample processing, data analysis and interpretation, write up, revising/editing.

**Logan:** Hypothesis development, data analysis and interpretation, write up, revising/editing, materials/funding.

**Folsom:** Blood collection, revising/editing.

**Bergeron:** Blood collection, revising/editing.

**Blackwell:** Hypothesis development, DNA extraction, revising/editing.

**Rowney:** DNA extraction, write up, revising/editing.

**Lukas:** Hypothesis development, data analysis and interpretation, write up, revising/editing, materials/funding.

###G. FUNDING

This research is funded by the Department of Human Behavior, Ecology and Culture at the Max Planck Institute for Evolutionary Anthropology.

###I. ACKNOWLEDGEMENTS

We thank Bronwyn Butcher and Irby Lovette at the Lab of Ornithology at Cornell University for providing the lab and training for processing the DNA samples using ddRADseq and for post-processing the raw data into a readily analyzable form.

###J. [REFERENCES](MyLibrary.bib)
